### ISSUE TITLE
Create a Sphinx theme based on the FOSSASIA theme

### ISSUE THREAD

**Issue by mariobehling** [fossasia/yaydoc#2]
Create a theme based on the FOSSASIA theme that can be used to generate docs site of our projects.

**Comment by imujjwal96** [comment]
Theme similar to http://blog.fossasia.org/  right?

**Comment by niranjan94** [comment]
@imujjwal96 No. The theme similar to 
- https://fossasia.org/
- https://2017.fossasia.org/
- http://codeheat.org/

You can also find their respective repositories

**Comment by imujjwal96** [comment]
I am not able to think of a way to theme a documentation like http://dev.loklak.org/server as those websites. The Blog's theme seems similar to it.   :innocent:

**Comment by niranjan94** [comment]
@imujjwal96 
The theme is bootstrap based. So, you can customize it. A doc site cannot look exactly like a conference site.
one way I could think of (off the top of my head) is ... You can keep the navbar as it is. Split the body into 2 column layout (3 wide column - 9 wide column). The side bar can have doc level nav. The main container can have content. 

The top nav can have project-level navigation. 

The possibilities are endless.

**Comment by sch00lb0y** [comment]
i was thinking changing the header and footer of the base theme at the beginning and we'll proceed further if any customization need

**Comment by imujjwal96** [comment]
in my opinion, we should create a theme from scratch. 
Also, we'll have to decide a wireframe for the documentations.

**Comment by pri22296** [comment]
We should inherit from the built in `basic` theme.

**Comment by sch00lb0y** [comment]
yeah because if we make from scratch then it'll be separate big project

**Comment by pri22296** [comment]
also because not using `basic` would lead to a lot of duplicating things already implemented. Almost all sphinx themes, custom one included inherit from `basic`

**Comment by imujjwal96** [comment]
Since Fossasia's theme is based on Bootstrap, we can extract this bootstrap theme for Sphinx to extend to Fossasia's documentations.

Resource: https://pypi.python.org/pypi/sphinx-bootstrap-theme/

**Comment by sch00lb0y** [comment]
@mariobehling @niranjan94  @pri22296 @imujjwal96  I extended the Bootstrap Sphinx theme to match Fosasia's Base theme.
Please checkout the demo version at : (https://sch00lb0y.github.io/tempCheck/) and give recommendations.

**Comment by pri22296** [comment]
Just for reference, [Here](https://ryan-roemer.github.io/sphinx-bootstrap-theme/sidebar.html) is how the page would have looked without @sch00lb0y's modifications.

**Comment by niranjan94** [comment]
@sch00lb0y how was this made ? Are you using the base stylesheets of the FOSSASIA theme or are you manually styling the default bootstrap theme to look like it ?

**Comment by sch00lb0y** [comment]
@niranjan94 i have used sulphur.css from codeheat repository

**Comment by sch00lb0y** [comment]
I have improved the theme. Please kindly check the [demo](https://sch00lb0y.github.io/tempCheck/)
@niranjan94 @pri22296 @imujjwal96 please give me your input

**Comment by pri22296** [comment]
![screenshot 17](https://cloud.githubusercontent.com/assets/11555869/26242573/80a65dc8-3ca6-11e7-8f48-c940a94199ca.png)

@sch00lb0y 
The topmost line is partially hidden by the menu bar.

**Comment by imujjwal96** [comment]
- The Fossasia's template has a fixed max-width (i guess it'd be 1600px).
- The main content has a white background, with the padding having color #eee


![image](https://cloud.githubusercontent.com/assets/8245662/26242765/30154ff8-3ca7-11e7-8774-8144bbbd4c7d.png)


Quoting @niranjan94:
> one way I could think of (off the top of my head) is ... You can keep the navbar as it is. Split the body into 2 column layout (3 wide column - 9 wide column). The side bar can have doc level nav. The main container can have content.

I agree with Niranjan and think the site would look more appealing if we have the sidebar or the side of the main content instead of being at the end of the docs.

**Comment by sch00lb0y** [comment]
@pri22296 @imujjwal96 i'll make changes. Thanks

**Comment by sch00lb0y** [comment]
@pri22296 @imujjwal96 [link](https://sch00lb0y.github.io/tempCheck/index.html) check this. i have added left sidebar and changed background to white 
Thanks!

**Comment by mariobehling** [comment]
This issue is not solved entirely.

**Comment by imujjwal96** [comment]
I was working on updating the FOSSASIA theme. There has been some update, but the theme that was taken initially now seems quite bloated. I was thinking if we could start-over from the most basic theme.
That would enable it for us to do a lot of customization. 

Also, I was thinking if we could use AMP in the theme. I think it would be an amazing addition to theming

**Comment by pri22296** [comment]
AMP? @sch00lb0y your views on this?

**Comment by imujjwal96** [comment]
@sch00lb0y your views on this?

**Comment by sch00lb0y** [comment]
AMP is a good idea but before going to AMP. we'll fix all the css bugs. once the theme is made properly and tested. we'll raise AMP as a separate issue and we'll work on it.

**Comment by imujjwal96** [comment]
If we are going to use AMP, then the whole HTML & CSS would have to be built from scratch.

**Comment by sch00lb0y** [comment]
there won't be any major change. we have to change some of the tag with AMP tags

**Comment by sch00lb0y** [comment]
yeah we can't use bootstrap but right now creating from the scratch won't be good idea, but still we can optimize out bootstrap version. check this [thread](https://stackoverflow.com/questions/38063272/using-accelerated-mobile-pages-with-bootstrap). @pri22296 please give your input regarding AMP

**Comment by pri22296** [comment]
AMP seems like a good idea. but can we use it with sphinx? I could not find any information regarding that.

**Comment by imujjwal96** [comment]
Ok! Let me try and make a sample site with sphinx and AMP. If it seems possible, we can integrate it.

**Comment by pri22296** [comment]
Sounds cool

**Comment by imujjwal96** [comment]
While working on it, I came to realize that amp uses `<amp-img>` in place of `<img>` tag
What can we do about this?

**Comment by sch00lb0y** [comment]
you have to make changes according to the AMP spec

**Comment by imujjwal96** [comment]
how can we change img to amp-img in sphinx?

**Comment by sch00lb0y** [comment]
I think you have to do by not extending the base theme, but creating a theme without extending from the base theme will not be a good idea

**Comment by pri22296** [comment]
This may not be achieved by a theme. AFAIK Sphinx uses docutils to generate html. I think we may need to write a custom HTML writer inheriting from the built in docutils HTML writing which uses AMP HTML tags instead of HTML tags.

**Comment by imujjwal96** [comment]
Should we drop the idea then?

**Comment by pri22296** [comment]
Let's keep it in our TodoList but let us first focus on the basic functionalities. AMP HTML certainly seems like a good idea but I don't think spending time on it now would be worth it. @sch00lb0y your views on this?

**Comment by sch00lb0y** [comment]
yeah, we'll keep on the list. our top priority should be script and theme.  Once we did with this we'll integrate yaydoc to all fossasia repository.

**Comment by imujjwal96** [comment]
OK.den
